lib/nimbase.h

```diff
@@ -459,3 +459,11 @@ typedef int Nim_and_C_compiler_disagree_on_target_architecture[sizeof(NI) == siz
 #elif defined(__FreeBSD__)
 #  include <sys/types.h>
 #endif
+
+/* Compile with -t:-DNIM_CHECK_ABI to enable */
```

Comment:
That's not how we do things. Make the #define unconditional and use -d:nimCheckAbi
Also this features needs to be documented in doc/nimc.rst